make-util.js

```diff
@@ -774,6 +774,136 @@ var validateTask = function (task) {
 exports.validateTask = validateTask;
 //------------------------------------------------------------------------------
 
+//------------------------------------------------------------------------------
+// Generate docs functions
+//------------------------------------------------------------------------------
+// Outputs a YAML snippet for the task
+var createYamlSnippet = function (taskJson, outFilePath) {
+    var outFile = fs.openSync(outFilePath, 'w');
+    var taskYaml = getTaskYaml(taskJson);
+    fs.writeSync(outFile, taskYaml);
+    fs.closeSync(outFile);
+}
+exports.createYamlSnippet = createYamlSnippet;
+
+function camelize(str) {
+    return str.replace(/(?:^\w|[A-Z]|\b\w|\s+)/g, function(match, index) {
+        return index == 0 ? match.toLowerCase() : match.toUpperCase();
+    });
+}
+
+var getAliasOrNameForInputName = function(inputs, inputName) {
+    var returnInputName = inputName;
+    inputs.forEach(function(input) {
+        if (input.name == inputName) {
+            if (input.aliases && input.aliases.length > 0) {
+                returnInputName = input.aliases[0];
+            }
+            else {
+                returnInputName = input.name;
+            }
+        }
+    });
+    return camelize(returnInputName);
+};
+
+var getInputAliasOrName = function(input) {
+    var returnInputName;
+    if (input.aliases && input.aliases.length > 0) {
+        returnInputName = input.aliases[0];
+    }
+    else {
+        returnInputName = input.name;
+    }
+    return camelize(returnInputName);
+};
+
+var cleanString = function(str) {
+    if (str) {
+        return str
+            .replace(/\r/g, '')
+            .replace(/\n/g, '')
+            .replace(/\"/g, '');
+    }
+    else {
+        return str;
+    }
+}
+
+var getTaskYaml = function(taskJson) {
+    var EOL = '\r\n';
+    var taskYaml = '';
+
+    taskYaml += '# ' + cleanString(taskJson.friendlyName) + EOL;
+    taskYaml += '# ' + cleanString(taskJson.description) + EOL;
```

Comment:
it looks like this will end up requiring horizontal scrolling in some (many?) cases, is that correct? just fyi, not saying this needs to be solved now but something to think about.

Reply:
Agreed.  More work is coming soon on the output format in general.